runCommandNoCC -> runCommand deprecation

roberth opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

Simplify to remove doubt when writing and reviewing expressions.

This produces no rebuilds except in the yarn2nix-moretea package, which is updated to use `runCommand`.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

roberth committed:
treewide: runCommandNoCC -> runCommand

This has been synonymous for ~5y.

roberth committed:
yarn2nix-moretea: Generate runCommand instead of -NoCC

This has been synonymous for ~5y.

roberth committed:
treewide: runCommandNoCC -> runCommand in generated code

This has been synonymous for ~5y.

Note that many of these runCommand bindings are unused, but that's
ok for generated code.

roberth committed:
runCommandNoCC: deprecate

It has been synonymous with runCommand ~5y.

roberth committed:
treewide: runCommandNoCCLocal -> runCommandLocal

It has always been a synonym.

roberth committed:
build-support/rust: remove unused runCommandNoCC

andir commented on pkgs/build-support/trivial-builders.nix:
```diff
@@ -24,16 +24,16 @@ rec {
   */
-  runCommand = runCommandNoCC;
-  runCommandLocal = runCommandNoCCLocal;
+  runCommandNoCC = lib.warn "runCommandNoCC is deprecated. Use the equivalent runCommand function instead." runCommand;
```
Do we plan removal at some point in the future? Now would be the time to announce that we will drop in in ~2022.11?

roberth replied:
I'd say it announces itself. No need to use up more brain bandwidth than necessary.

siraben replied:
Though, maybe link to the PR like we did for deprecating `stdenv.lib`. Not necessary though.

roberth committed:
listadmin: fix deprecation

roberth committed:
nixpkgs-basic-release-checks: Check without aliases

This allows the aliases file to produce deprecation warnings
without triggering the check against warnings. This allows
for a better user experience when deprecations occur.

Moreover, Nixpkgs itself shouldn't rely on its backcompat aliases.

siraben commented:
Result of `nixpkgs-review pr 134225` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>yarn2nix</li>
  </ul>
</details>

roberth commented:
Oh no, I was still looking into making ofborg accept this.

roberth commented:
I'll make a PR to fix ofborg eval.

roberth commented:
The deprecation warning has been reverted in #134230, making this PR ineffective.

We can't have deprecate it until https://github.com/NixOS/ofborg/issues/575, unless someone knows a workaround to please / trick ofborg.
